sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/accessibility/DomElementsShouldUseAutocompleteAttributeCorrectlyCheck.java

```diff
@@ -0,0 +1,323 @@
+/*
+ * SonarSource HTML analyzer :: Sonar Plugin
+ * Copyright (c) 2010-2024 SonarSource SA and Matthijs Galesloot
+ * sonarqube@googlegroups.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sonar.plugins.html.checks.accessibility;
+
+import org.sonar.check.Rule;
+import org.sonar.plugins.html.api.HtmlConstants;
+import org.sonar.plugins.html.api.accessibility.ControlGroup;
+import org.sonar.plugins.html.checks.AbstractPageCheck;
+import org.sonar.plugins.html.node.TagNode;
+
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+@Rule(key = "S6840")
+public class DomElementsShouldUseAutocompleteAttributeCorrectlyCheck extends AbstractPageCheck {
```

Comment:
Let's rename the check to something that doesn't look like a sentence or sound like the name of a test case :)
I suggest naming it `ValidAutocompleteCheck` (or at least something shorter).

Reply:
I actually wondered if there was a naming convention for rules - `SXXXXCheck`, a sentence (like here), something descriptive, something imperative...

I'll apply your suggestion, and it would be cool if we could agree on a convention for future additions.

Reply:
I don't think we have any convention, maybe we should. In general, we don't pay much attention to the check names since they are internal, but in your case, it was hard to miss it ^^